### PR TITLE
[prometheus-node-exporter] Add Support for Environment Variables for KubeRBACProxy Container

### PR DESCRIPTION
**Pull request by nfrazier08**
#### What this PR does / why we need it
- A variable was also added to configure env variable for the KubeRBACProxy container

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by nfrazier08** [comment]
CC: @zanhsieh @zeritti @gianrubio

**Review by zanhsieh — APPROVED** [review]

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @nfrazier08, for your PR. Please, see my comments below.

File: charts/prometheus-node-exporter/values.yaml [hunk 1]
```diff
@@ -39,6 +39,9 @@ global:
 # The requests are served through the same service but requests are HTTPS.
 kubeRBACProxy:
   enabled: false
+  env: {}
+  ##  env:
+  ##    VARIABLE: value
```
[the remark this answers is not in the document]
Comment (nfrazier08) [reply charts/prometheus-node-exporter/values.yaml:44]: Changed.

File: charts/prometheus-node-exporter/templates/daemonset.yaml [hunk 2]
```diff
@@ -230,6 +230,13 @@ spec:
           terminationMessagePolicy: {{ .terminationMessagePolicy }}
           {{- end }}
           {{- end }}
+          {{- with .Values.kubeRBACProxy.env }}
+          env:
+            {{- range $key, $value := .Values.kubeRBACProxy.env }}
```
[the remark this answers is not in the document]
Comment (nfrazier08) [reply charts/prometheus-node-exporter/templates/daemonset.yaml:235]: Strange - I templated this before I pushed it and didn't get the nil interface error. Perhaps I accidentally deleted the `$`. Either way, I made the fix.

**Review by zeritti — APPROVED** [review]

**Pull request merged**